### PR TITLE
Add an additional interpolation test...

### PR DESCRIPTION
**Pull request by foolswood**
and fix the issues it found.

Addresses some of #6.

### CODE REVIEW

**Comment by ch3pjw** [comment]
So... I might need you to walk me through some of this!

File: types/test/test_interpolation.c [hunk 1]
```diff
@@ -297,8 +308,42 @@ static rage_Error interpolator_immediate_change_test() {
...
+            return rv;
+    }
+    rage_Finaliser * change_complete = change_float_ts_to(interpolator, 1.0, 2);
+    for (uint8_t i = 0; i < 2; i++) {
+        rage_InterpolatedView * v = rage_interpolator_get_view(interpolator, i);
+        rage_interpolated_view_advance(v, 1);
+        rage_InterpolatedValue const * r = rage_interpolated_view_value(v);
+        rv = check_float_value(r, 0.0, 1);
+        if (RAGE_FAILED(rv))
+            return rv;
```
Comment (foolswood) [inline types/test/test_interpolation.c]: replace with break

File: types/src/interpolation.c [hunk 2]
```diff
@@ -267,6 +267,19 @@ rage_InterpolatedValue const * rage_interpolated_view_value(
             view->value.value + i, start->value + i, end->value + i,
             view->pos - start->frame, duration);
     }
+    // FIXME: this is ugly because the interpolators set it "wrong"
```
Comment (foolswood) [inline types/src/interpolation.c]: confusing fixme: explain that UINT32_MAX is "forever"

Comment (foolswood) [reply types/src/interpolation.c]: valid_for should be a uint32_t

File: types/src/interpolation.c [hunk 3]
```diff
@@ -267,6 +267,19 @@ rage_InterpolatedValue const * rage_interpolated_view_value(
...
+    if (duration == UINT32_MAX)
+        view->value.valid_for = duration;
+    // Take into account that timeseries might have changed in the future
+    // FIXME: a lot of commonality with seek!
+    rage_FrameSeries * active_pts = atomic_load_explicit(
+        &view->interpolator->points, memory_order_consume);
+    if (active_pts != view->points) {
+        uint32_t next_view_valid_from = (view->interpolator->valid_from > view->pos) ?
```
Comment (foolswood) [inline types/src/interpolation.c]: Should be able to simplify as should always be in the future.

**Review by ch3pjw — APPROVED** [review]

**Pull request merged**